nautobot/core/jobs/__init__.py

```diff
@@ -0,0 +1,115 @@
+import os
+
+from django.conf import settings
+
+from nautobot.core.celery import register_jobs
+from nautobot.extras.choices import LogLevelChoices
+from nautobot.extras.datasources import (
+    ensure_git_repository,
+    git_repository_dry_run,
+    refresh_datasource_content,
+)
+from nautobot.extras.jobs import Job, ObjectVar
+from nautobot.extras.models import GitRepository
+
+
+name = "System Jobs"
+
+
+class GitRepositoryPullAndRefreshData(Job):
+    """
+    System job to clone and/or pull a Git repository, then invoke refresh_datasource_content().
+    """
+
+    repository = ObjectVar(
+        description="Git Repository to pull and refresh",
+        label="Git Repository",
+        model=GitRepository,
+    )
+
+    class Meta:
+        name = "Git Repository: Sync"
+        has_sensitive_variables = False
+
+    def run(self, repository):
+        job_result = self.get_job_result()
+        user = job_result.user
+
+        job_result.log(
+            f'Creating/refreshing local copy of Git repository "{repository.name}"...',
+            logger=self.logger,
+        )
+
+        try:
+            if not os.path.exists(settings.GIT_ROOT):
+                os.makedirs(settings.GIT_ROOT)
+
+            ensure_git_repository(
+                repository,
+                job_result=job_result,
+                logger=self.logger,
+            )
+
+            job_result.log(
+                f'The current Git repository hash is "{repository.current_head}"',
+            )
+
+            refresh_datasource_content("extras.gitrepository", repository, user, job_result, delete=False)
+
+        except Exception as exc:
+            job_result.log(
+                f"Error while refreshing {repository.name}: {exc}",
+                level_choice=LogLevelChoices.LOG_FAILURE,
+                logger=self.logger,
+            )
+            raise
```

Comment:
This *might* be redundant - `ensure_git_repository` already catches, logs, and re-raises likely exceptions that occur within it, and `refresh_datasource_content` catches and logs (but does not re-raise) likely exceptions there as well.